### PR TITLE
boards/nucleo-g071rb: add initial support

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR adds support for the nucleo-g071rb. This board is almost the same as the nucleo-g07rb but the CPU provides LPTIM1 (and LPTIM2) so the rtt peripheral is adapted to work on this CPU. 

### Testing procedure

Run `BUILD_IN_DOCKER=1 ./dist/tools/compile_and_test_for_board/compile_and_test_for_board.py . nucleo-g071rb --jobs=8 --with-test-only`

<details><summary>Results</summary>

```
Failures during compilation:
- [tests/suit_manifest](tests/suit_manifest/compilation.failed)

Failures during test:
- [tests/driver_at86rf2xx_aes](tests/driver_at86rf2xx_aes/test.failed)
- [tests/driver_grove_ledbar](tests/driver_grove_ledbar/test.failed)
- [tests/driver_my9221](tests/driver_my9221/test.failed)
- [tests/emcute](tests/emcute/test.failed)
- [tests/gnrc_dhcpv6_client](tests/gnrc_dhcpv6_client/test.failed)
- [tests/gnrc_dhcpv6_client_6lbr](tests/gnrc_dhcpv6_client_6lbr/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_ipv6_ext_opt](tests/gnrc_ipv6_ext_opt/test.failed)
- [tests/gnrc_ipv6_nib_dns](tests/gnrc_ipv6_nib_dns/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/periph_timer_short_relative_set](tests/periph_timer_short_relative_set/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
- [tests/pkg_semtech-loramac](tests/pkg_semtech-loramac/test.failed)
- [tests/pkg_tensorflow-lite](tests/pkg_tensorflow-lite/test.failed)
```

=> no big surprise here

</details>

### Issues/PRs references

None

### CODE REVIEW

File: cpu/stm32/periph/rtt_all.c [hunk 1]
```diff
@@ -77,10 +77,13 @@ register. */
...
+#elif defined(CPU_FAM_STM32G0)
+#define IMR_REG             IMR1
+#define EXTI_IMR_BIT        EXTI_IMR1_IM29
 #elif defined(CPU_FAM_STM32G4)
 #define IMR_REG             IMR2
 #define EXTI_IMR_BIT        EXTI_IMR2_IM37
-#elif defined(CPU_FAM_STM32L0) || defined(CPU_FAM_STM32G0)
+#elif defined(CPU_FAM_STM32L0)
 #define IMR_REG             IMR
 #define EXTI_IMR_BIT        EXTI_IMR_IM29
```
Comment (benpicco) [inline cpu/stm32/periph/rtt_all.c:88]: I was wondering how this wasn't caught by `nucleo-g070rb` but it turns out this board does not advertise the `periph_rtt` feature.

Should we add this there as well?

Comment (aabadie) [reply cpu/stm32/periph/rtt_all.c:88]: There's no LPTIM1 on g070 :)

**Review by benpicco — APPROVED** [review]
Test results are provided. also fixes rtt for g0 line of MCUs.

**Comment by aabadie** [comment]
That was fast, thanks :)

**Pull request merged**